### PR TITLE
Animations containers

### PR DESCRIPTION
**Pull request by Arthi-chaud**
Implementation of Animations container to easily load them from a file, apply them to a model, which can then be drawn.

The ModelAnimations class is responsible for loading/unloading all data
The ModelAnimation class is an encapsulation of raylib's. It's used to access data related to the animation.

### CODE REVIEW

File: lib/Ray/sources/Drawables/2D/Triangle.cpp [hunk 1]
```diff
@@ -52,9 +52,8 @@ void RAY::Drawables::Drawables2D::Triangle::drawOn(RAY::Window &)
 	DrawTriangle(this->_position, this->_posB, this->_posC, this->_color);
 }
 
-void RAY::Drawables::Drawables2D::Triangle::drawOn(RAY::Image &image)
+void RAY::Drawables::Drawables2D::Triangle::drawOn(RAY::Image &)
 {
-	(void)image;
 	throw std::exception();
```
Comment (AnonymusRaccoon) [inline lib/Ray/sources/Drawables/2D/Triangle.cpp:57]: You should throw a specific exception. That means a derived of std::runtime_exception with a specifc type and a specfic error message.

Comment (AnonymusRaccoon) [reply lib/Ray/sources/Drawables/2D/Triangle.cpp:57]: Here that could be a NotSupportedError with a message of "A triangle can't be drawn to an image".

File: lib/Ray/sources/Model/Model.hpp [hunk 2]
```diff
@@ -38,12 +46,29 @@ namespace RAY {
...
 			//! @brief Unload model (including meshes) from memory (RAM and/or VRAM)
-			bool unload();
+			bool unload() override;
 
 			//! @brief Unload model (excluding meshes) from memory (RAM and/or VRAM)
 			bool unloadKeepMeshes();
+
+			//! @brief Update model animation pose
+			//! @return false if animation is not compatible
+			bool setAnimation(const RAY::ModelAnimation &animation);
```
Comment (AnonymusRaccoon) [inline lib/Ray/sources/Model/Model.hpp:56]: In C++ errors should be reported via exceptions, not via the return type.

**Review by Octopus773 — DISMISSED** [review]

**Comment by Octopus773** [comment]
You should push your ressources folder

**Review by Octopus773 — APPROVED** [review]

**Pull request merged**